cpp/foxglove/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+cmake_minimum_required(VERSION 3.25)
+project(foxglove-sdk)
+
+include(CTest)
+include(FetchContent)
+
+FetchContent_Declare(
+  Catch2
+  GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+  GIT_TAG        v3.8.0
+)
+FetchContent_MakeAvailable(Catch2)
+list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/extras)
+include(Catch)
+
+add_library(foxglove_c STATIC IMPORTED)
+set_property(TARGET foxglove_c PROPERTY IMPORTED_LOCATION_DEBUG ${CMAKE_CURRENT_SOURCE_DIR}/../../target/debug/libfoxglove_c.a)
+set_property(TARGET foxglove_c PROPERTY IMPORTED_LOCATION_RELEASE ${CMAKE_CURRENT_SOURCE_DIR}/../../target/release/libfoxglove_c.a)
+set_property(TARGET foxglove_c PROPERTY IMPORTED_LOCATION_RELWITHDEBINFO ${CMAKE_CURRENT_SOURCE_DIR}/../../target/release/libfoxglove_c.a)
+target_include_directories(foxglove_c INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/../../c/foxglove/generated/include)
+
+file(GLOB_RECURSE foxglove_srcs CONFIGURE_DEPENDS
+    "include/*.hpp"
+    "src/*.hpp"
+    "src/*.cpp"
+)
+
+add_library(foxglove STATIC "${foxglove_srcs}")
+set_property(TARGET foxglove PROPERTY CXX_STANDARD 20)
```

Comment:
Are we using features which require C++20? I would have expected C++14 would be sufficient.

Reply:
2020 was already 5 years ago, do you think it's still too early to move forward with it?

Reply:
🤷 my experience is c/c++ moves a lot slower. Would having this make it hard to incorporate an older library into a build, or is it possible to mix and match?

When I was building the C++ runtime of a project a couple years ago, I remember deciding on 14 because it included enough of the modern stuff I wanted, and there weren't major features after that I actually needed.

Reply:
(this is as opposed to cmake version, where I found a huge benefit to requiring newer cmake)

Reply:
I think 17 is a good compromise